beacon_node/http_api/src/block_id.rs

```diff
@@ -266,6 +267,29 @@ impl BlockId {
             Err(e) => Err(warp_utils::reject::beacon_chain_error(e)),
         }
     }
+
+    pub async fn blob_sidecar_list_indexed<T: BeaconChainTypes>(
+        &self,
+        indices: BlobIndicesQuery,
+        chain: &BeaconChain<T> 
+    ) -> Result<BlobSidecarList<T::EthSpec>, warp::Rejection> {
+        let blob_sidecar_list = self.blob_sidecar_list(&chain).await?;
+        let blob_sidecar_list_indexed = match indices.indices {
+            Some(vec) => {
+                let list = blob_sidecar_list.into_iter()
+                .enumerate()
+                .filter(|(index, _)| vec.contains(index))
+                .map(|(_, blob)| blob)
+                .collect();
+                let indexed_list = BlobSidecarList::new(list);
+                indexed_list.unwrap()
```

Comment:
Although this is an unexpected scenario, we should always avoid using `unwrap` in production code. In this case it's probably best to return a server error, something like:

```rust
BlobSidecarList::new(list)
                    .map_err(|e| warp_utils::reject::custom_server_error(format!("{e:?}")))?
```
The `map_err` function maps the ssz `Error` type into the required `warp::Rejection` error type, and the `?` allow the error to be automatically propagated / returned.